core/Updates/3.0.0-b2.php

```diff
@@ -21,4 +23,84 @@ public static function isMajorUpdate()
     {
         return true;
     }
+
+    /**
+     * @var MigrationFactory
+     */
+    private $migration;
+
+    /**
+     * @var string
+     */
+    private $userTable = 'user';
+
+    public function __construct(MigrationFactory $factory)
+    {
+        $this->migration = $factory;
+    }
+
+    /**
+     * Here you can define one or multiple SQL statements that should be executed during the update.
+     * @param Updater $updater
+     * @return Migration[]
+     */
+    public function getMigrations(Updater $updater)
+    {
+        $migrations = [];
+        $migrations = $this->getUserDatabaseMigrations($migrations);
+
+        return $migrations;
+    }
+
+
+    public function doUpdate(Updater $updater)
+    {
+        $updater->executeMigrations(__FILE__, $this->getMigrations($updater));
+
+        $this->hashUserPasswords();
+    }
+
+      /**
+     * @param Migration[] $queries
+     * @return Migration[]
+     */
+    private function getUserDatabaseMigrations($queries)
+    {
+        $queries[] = $this->migration->db->changeColumn($this->userTable, 'password', 'password', 'VARCHAR(255) NOT NULL');
+        $queries[] = $this->migration->db->addColumn($this->userTable, 'password_hashed', 'VARCHAR(255) NOT NULL', 'password');
+        $queries[] = $this->migration->db->changeColumn($this->userTable, 'token_auth', 'token_auth', 'VARCHAR(64) NOT NULL');
+
+        return $queries;
+    }
+
+    /**
+     * Hashes existing user passwords with bcrypt around current MD5.
+     */
+    private function hashUserPasswords()
+    {
+        $db            = Db::get();
+        $optionPattern = '%s_legacy_password';
+        $userTable     = Common::prefixTable($this->userTable);
+
+        $users = $db->fetchAll(
+            'SELECT `login`, `password` FROM `' . $userTable . '` WHERE LENGTH(`password`) = 32'
+        );
+
+        foreach ($users as $user) {
+            $crypted = password_hash($user['password'], PASSWORD_BCRYPT);
+
+            $db->query(
+                'UPDATE `' . $userTable . '`'
+                . ' SET `password` = :password, `password_hashed` = :password_hashed'
```

Comment:
Will there be also an update for `token_auth = hash('sha256, token_auth)`? `sha2` was only added recently in MySQL 5.5 so we might not be able to use it actually. Will need to do it in PHP unfortunately